ngraph/frontend/onnx/frontend/src/place.cpp

```diff
@@ -209,3 +248,133 @@ Place::Ptr PlaceOpONNX::get_input_port(const std::string& input_name) const {
     }
     return nullptr;
 }
+
+std::vector<Place::Ptr> PlaceOpONNX::get_consuming_ports() const {
+    std::vector<Place::Ptr> consuming_ports;
+    const auto out_ports_size = m_editor->get_output_ports(m_node).size();
+    for (int out_idx = 0; out_idx < out_ports_size; ++out_idx) {
+        auto consuming_ops_out = get_output_port(out_idx)->get_consuming_ports();
+        consuming_ports.insert(consuming_ports.end(), consuming_ops_out.begin(), consuming_ops_out.end());
+    }
+    return consuming_ports;
+}
+
+namespace {
+std::vector<Place::Ptr> get_consuming_ops(std::vector<Place::Ptr> input_ports) {
+    std::vector<Place::Ptr> consuming_ops;
+    std::transform(std::begin(input_ports),
+                   std::end(input_ports),
+                   std::back_inserter(consuming_ops),
+                   [](const Place::Ptr place) {
+                       return place->get_consuming_operations().at(0);
+                   });
+
+    return consuming_ops;
+}
+}  // namespace
+
+std::vector<Place::Ptr> PlaceOpONNX::get_consuming_operations() const {
+    std::vector<Place::Ptr> consuming_ports = get_consuming_ports();
+    return get_consuming_ops(consuming_ports);
+}
+
+std::vector<Place::Ptr> PlaceOpONNX::get_consuming_operations(int output_port_index) const {
+    std::vector<Place::Ptr> consuming_ports = get_output_port(output_port_index)->get_consuming_ports();
+    return get_consuming_ops(consuming_ports);
+}
+
+std::vector<Place::Ptr> PlaceOpONNX::get_consuming_operations(const std::string& output_port_name) const {
+    std::vector<Place::Ptr> consuming_ports = get_output_port(output_port_name)->get_consuming_ports();
+    return get_consuming_ops(consuming_ports);
+}
+
+Place::Ptr PlaceOpONNX::get_producing_operation() const {
+    const auto input_port = get_input_port();
+    if (input_port != nullptr) {
+        return input_port->get_producing_operation();
+    }
+    return nullptr;
+}
+
+Place::Ptr PlaceOpONNX::get_producing_operation(int input_port_index) const {
+    const auto input_port = get_input_port(input_port_index);
+    if (input_port != nullptr) {
+        return input_port->get_producing_operation();
+    }
+    return nullptr;
+}
+
+Place::Ptr PlaceOpONNX::get_producing_operation(const std::string& input_port_name) const {
+    const auto input_port = get_input_port(input_port_name);
+    if (input_port != nullptr) {
+        return input_port->get_producing_operation();
+    }
+    return nullptr;
+}
+
+bool PlaceOpONNX::is_equal(Place::Ptr another) const {
+    if (const auto place_op = std::dynamic_pointer_cast<PlaceOpONNX>(another)) {
+        const auto& another_node = place_op->get_editor_node();
+        if (m_editor->is_correct_and_unambiguous_node(m_node) ||
+            m_editor->is_correct_and_unambiguous_node(another_node)) {
```

Comment:
this 'or' looks not covered by tests. I tried python's one and onnx_frontend_tests

Reply:
It is hard to check now because in the current state we can create an operation  with set name only using `InputModelONNX::get_place_by_operation_name`, but we are also checking `is_correct_and_unambiguous_node` there: https://github.com/openvinotoolkit/openvino/blob/8598ce89a342ce39a1bd0b13a0603fc276ce26c5/ngraph/frontend/onnx/frontend/src/input_model.cpp#L62.
I think that we can add such test when `set_name_for_operation` will be implemented. I've left the comment about it in 65770